### ISSUE TITLE
Strange buffer_hunk_preview behaviour

### ISSUE THREAD

**Issue by timsofteng** [tanvirtin/vgit.nvim#137]
### Description

When I open hunk preview of my change in specific place of file I expect to see just this piece of code.
Instead of it `buffer_hunk_preview` function now open full buffer.
Take a look into screencast please

https://user-images.githubusercontent.com/51912173/135091581-864de6da-05f1-4ea9-83d2-8840b3dccbaf.mp4

I believe we need just window with only changing of current place of the file.

- nvim --version
NVIM v0.5.1
Build type: Release
LuaJIT 2.1.0-beta3
- Void linux:

**Comment by tanvirtin** [comment]
Hi, some of the functionalities are expected to change to shape the plugin a bit better. Once the first version of `VGit` is released expect the features to stablitize more.

I have made some changes to buffer_hunk_preview, to give it a feel closer to the VSCode git integration, from which this plugin heavily derives ideas from, for example:

![Capture](https://user-images.githubusercontent.com/25164326/135165408-2e47318c-7d91-4d57-8f5c-abe61d25479d.PNG)

Having an entire buffer is expected inside the hunk preview. When you open the preview however, it will always focus on the current hunk the user is on, so you always get the quick hunk preview. The outcome has not changed. Having an entire buffer will give the user more context of the changes below and above the hunk.

I will be adding text folding for non hunks in the future, if I feel it to be more useful, for now I have set a fixed height for the hunk that can be customized.
